packages/edit-post/src/components/layout/index.js

```diff
@@ -167,9 +161,20 @@ function Layout( { styles } ) {
 		},
 		[ entitiesSavedStatesCallback ]
 	);
-	const [ inserterDialogRef, inserterDialogProps ] = useDialog( {
-		onClose: () => setIsInserterOpened( false ),
-	} );
+
+	const secondarySidebar = () => {
```

Comment:
Why is this a function and not directly the element.

Reply:
Quirk of `InterfaceSkeleton`:

https://github.com/WordPress/gutenberg/blob/755a2272cf0985e5b56978cb506fd6522353c6a0/packages/interface/src/components/interface-skeleton/index.js#L98-L107

The `secondarySidebar` is always rendered unless falsy.
If we handled the conditional render in the components themselves, the `secondarySidebar` prop would be a (truthy) component rendering `null`, which would basically display an empty sidebar.

Reply:
I don't think so, you're calling the function here https://github.com/WordPress/gutenberg/pull/31047/files#diff-8933c0b377d83066f7dde9c8b573bab9a7bfe60f73919e4e5e80aff92a88800cR199 so it's the exact same thing for me.

Reply:
Ooh, I see what you mean, like a `<SecondarySidebar />` component.
It can be done, yes.

I'm currently working on a PR containing a few small improvements on the Site Editor's list view, and will include the change there and eventually backport it here.

Reply:
Or actually, no? 😄 

I've tried something like this:

```js
function SecondarySidebar() {
  if ( isInserterOpen ) return <InserterSidebar />;
  if ( isListViewOpen ) return <ListViewSidebar />;
  return null;
}

// ...

<InterfaceSkeleton secondarySidebar={ <SecondarySidebar /> } />
```

In this PR, we are using the return value of the function as `secondarySidebar` prop, which can be either a component or null.
In the above example instead, `secondarySidebar` is always a component, which means `InterfaceSkeleton` will always render the `div.interface-interface-skeleton__secondary-sidebar`.

On large screens, this is negligible: the sidebar has dynamic width, and without any content it just displays as `width: 0`.
On the other hand, on small screens, the sidebar fills the entire screen, whether it has or doesn't have any content inside. Basically, it would show a white screen.

Reply:
I think you're probably misunderstanding my comment :)  

I mean something like this:

```
let secondarySidebar = null;
if ( mode === 'visual' && isInserterOpened ) {
 	secondarySidebar = <InserterSidebar />;
 }
 if ( mode === 'visual' && isListViewOpened ) {
	secondarySidebar =  (
		<AsyncModeProvider value="true">
			<ListViewSidebar />
		</AsyncModeProvider>
 	);
 }
```

Reply:
Ah, ok then it's just a matter of style. 😄 
I'm personally not very fond of `let`, and I find early returns more readable.